linking error: undefined reference to symbol: pthread_...

I copied in build_config.hh to continue the build (#13), now failing with the following error (on Arch Linux):

```
[38/48] Building C object CMakeFiles/astroi...ir/src/utils/gmime/gmime-filter-html-bq.c.o
../src/utils/gmime/gmime-filter-html-bq.c: In function ‘g_mime_filter_html_bq_new’:
../src/utils/gmime/gmime-filter-html-bq.c:583:2: warning: ‘g_object_newv’ is deprecated: Use 'g_object_new_with_properties' instead [-Wdeprecated-declarations]
  filter = g_object_newv (GMIME_TYPE_FILTER_HTML_BQ, 0, NULL);
  ^~~~~~
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/gmime-3.0/gmime/gmime-filter.h:26,
                 from /usr/include/gmime-3.0/gmime/gmime-filter-html.h:26,
                 from ../src/utils/gmime/gmime-filter-html-bq.c:33:
/usr/include/glib-2.0/gobject/gobject.h:429:13: note: declared here
 gpointer    g_object_newv        (GType           object_type,
             ^~~~~~~~~~~~~
[48/48] Linking CXX executable astroid
FAILED: astroid
: && /usr/bin/c++  -std=c++14   CMakeFiles/astroid.dir/src/account_manager.cc.o CMakeFiles/astroid.dir/src/astroid.cc.o CMakeFiles/astroid.dir/src/chunk.cc.o CMakeFiles/astroid.dir/src/command_bar.cc.o CMakeFiles/astroid.dir/src/compose_message.cc.o CMakeFiles/astroid.dir/src/config.cc.o CMakeFiles/astroid.dir/src/crypto.cc.o CMakeFiles/astroid.dir/src/db.cc.o CMakeFiles/astroid.dir/src/main.cc.o CMakeFiles/astroid.dir/src/main_window.cc.o CMakeFiles/astroid.dir/src/message_thread.cc.o CMakeFiles/astroid.dir/src/poll.cc.o CMakeFiles/astroid.dir/src/modes/edit_message.cc.o CMakeFiles/astroid.dir/src/modes/forward_message.cc.o CMakeFiles/astroid.dir/src/modes/help_mode.cc.o CMakeFiles/astroid.dir/src/modes/keybindings.cc.o CMakeFiles/astroid.dir/src/modes/log_view.cc.o CMakeFiles/astroid.dir/src/modes/mode.cc.o CMakeFiles/astroid.dir/src/modes/paned_mode.cc.o CMakeFiles/astroid.dir/src/modes/raw_message.cc.o CMakeFiles/astroid.dir/src/modes/reply_message.cc.o CMakeFiles/astroid.dir/src/modes/saved_searches.cc.o CMakeFiles/astroid.dir/src/modes/editor/external.cc.o CMakeFiles/astroid.dir/src/modes/thread_index/query_loader.cc.o CMakeFiles/astroid.dir/src/modes/thread_index/thread_index.cc.o CMakeFiles/astroid.dir/src/modes/thread_index/thread_index_list_cell_renderer.cc.o CMakeFiles/astroid.dir/src/modes/thread_index/thread_index_list_view.cc.o CMakeFiles/astroid.dir/src/modes/thread_view/dom_utils.cc.o CMakeFiles/astroid.dir/src/modes/thread_view/theme.cc.o CMakeFiles/astroid.dir/src/modes/thread_view/thread_view.cc.o CMakeFiles/astroid.dir/src/modes/thread_view/web_inspector.cc.o CMakeFiles/astroid.dir/src/actions/action.cc.o CMakeFiles/astroid.dir/src/actions/action_manager.cc.o CMakeFiles/astroid.dir/src/actions/cmdaction.cc.o CMakeFiles/astroid.dir/src/actions/difftag_action.cc.o CMakeFiles/astroid.dir/src/actions/onmessage.cc.o CMakeFiles/astroid.dir/src/actions/tag_action.cc.o CMakeFiles/astroid.dir/src/actions/toggle_action.cc.o CMakeFiles/astroid.dir/src/utils/address.cc.o CMakeFiles/astroid.dir/src/utils/cmd.cc.o CMakeFiles/astroid.dir/src/utils/date_utils.cc.o CMakeFiles/astroid.dir/src/utils/gravatar.cc.o CMakeFiles/astroid.dir/src/utils/resource.cc.o CMakeFiles/astroid.dir/src/utils/ustring_utils.cc.o CMakeFiles/astroid.dir/src/utils/utils.cc.o CMakeFiles/astroid.dir/src/utils/vector_utils.cc.o CMakeFiles/astroid.dir/src/utils/gmime/gmime-compat.cc.o CMakeFiles/astroid.dir/src/utils/gmime/gmime-filter-html-bq.c.o CMakeFiles/astroid.dir/src/utils/gmime/gtrie.c.o CMakeFiles/astroid.dir/src/utils/gmime/url-scanner.c.o CMakeFiles/astroid.dir/src/modes/editor/editor.cc.o CMakeFiles/astroid.dir/src/modes/editor/plugin.cc.o  -o astroid  -Wl,-rpath,/usr/local/lib /usr/local/lib/libnotmuch.so -lwebkitgtk-3.0 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -ljavascriptcoregtk-3.0 -lglib-2.0 -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lgio-2.0 -lcairomm-1.0 -lcairo -lsigc-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -lglibmm-2.4 -lgobject-2.0 -lglib-2.0 -lsigc-2.0 -lgmime-3.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lvte-2.91 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lz -lpcre2-8 -lgnutls -lsass /usr/lib/libboost_filesystem.so /usr/lib/libboost_program_options.so /usr/lib/libboost_log_setup.so /usr/lib/libboost_log.so /usr/lib/libboost_thread.so /usr/lib/libboost_date_time.so /usr/lib/libboost_system.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_atomic.so -lsoup-2.4 -ljavascriptcoregtk-3.0 -lgtkmm-3.0 -latkmm-1.6 -lgdkmm-3.0 -lgiomm-2.4 -lpangomm-1.4 -lglibmm-2.4 -lcairomm-1.0 -lsigc-2.0 -lgmime-3.0 -lvte-2.91 -lz -lpcre2-8 -lgnutls -lsass /usr/lib/libboost_filesystem.so /usr/lib/libboost_program_options.so /usr/lib/libboost_log_setup.so /usr/lib/libboost_log.so /usr/lib/libboost_thread.so /usr/lib/libboost_date_time.so /usr/lib/libboost_system.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_atomic.so && :
/usr/bin/ld: CMakeFiles/astroid.dir/src/astroid.cc.o: undefined reference to symbol 'pthread_rwlock_wrlock@@GLIBC_2.2.5'
/usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,8 @@ pkg_check_modules (GMIME3     REQUIRED  gmime-3.0>=3.0.0)
 pkg_check_modules (WEBKITGTK3 REQUIRED  webkitgtk-3.0)
 pkg_check_modules (SASS       REQUIRED  libsass)
 
+find_package ( Threads )
+
 find_package ( Boost REQUIRED
   COMPONENTS
   filesystem
@@ -93,7 +95,6 @@ add_compile_options (
 
 add_compile_options (
   -Wall
-  -pthread
   -g
   -Wextra
   )
@@ -216,6 +217,7 @@ target_link_libraries (
   ${VTE2_LIBRARIES}
   ${SASS_LIBRARIES}
   ${Boost_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
   )
 
 ##

